WordPress/src/main/java/org/wordpress/android/GCMMessageService.java

```diff
@@ -122,20 +121,19 @@ private void handleDefaultPush(String from, @NonNull Bundle data) {
 
     private void trySaveNoteIfNotAlreadyInBucket(Bundle data) {
         if (SimperiumUtils.getNotesBucket() != null) {
-            Note note = null;
             String noteId = data.getString(PUSH_ARG_NOTE_ID, "");
             try {
-                note = SimperiumUtils.getNotesBucket().get(noteId);
+                SimperiumUtils.getNotesBucket().get(noteId);
                 // all good if we got here
             } catch (BucketObjectMissingException e) {
                 AppLog.e(T.NOTIFS, e);
                 SimperiumUtils.trackBucketObjectMissingWarning(e.getMessage(), noteId);
 
-                if (data != null && data.containsKey(PUSH_ARG_NOTE_FULL_DATA)) {
+                if (data.containsKey(PUSH_ARG_NOTE_FULL_DATA)) {
```

Comment:
I agree with you, but `data` is already accessed at line 124 without a null check ;)

I will add null check there for security reason.